npm/grep/test.spec.js

```diff
@@ -0,0 +1,288 @@
+const Fixtures = require('@tooling/system-tests')
+const FixturesScaffold = require('@tooling/system-tests/lib/dep-installer')
+const execa = require('execa')
+const path = require('path')
+const fs = require('fs-extra')
+
+const scaffoldGrepProject = async () => {
+  const project = 'grep'
+  const projectPath = Fixtures.projectPath(project)
+
+  Fixtures.removeProject(project)
+  await Fixtures.scaffoldProject(project)
+  await FixturesScaffold.scaffoldProjectNodeModules({ project })
+
+  return projectPath
+}
+
+const runCommandInProject = async (command, projectPath, _args) => {
+  const [ex, ...args] = command.split(' ')
+
+  // _args is used for scenarios where we are filtering with spaces
+  return await execa(ex, _args ? _args : args, { cwd: projectPath, stdio: 'inherit' })
+}
+
+const cypressGrepPackagePath = path.join(__dirname)
+
+describe('@cypress/grep', () => {
+  let projectPath
+
+  before(async () => {
+    projectPath = await scaffoldGrepProject(cypressGrepPackagePath)
+
+    await runCommandInProject(`yarn add @cypress/grep@file:${cypressGrepPackagePath}`, projectPath)
+  })
+
+  it('run Unit Tests', async () => {
+    await runCommandInProject('npx cypress run --spec cypress/e2e/unit.js --config specPattern="**/unit.js"', projectPath)
+  })
+
+  it('Run e2e tests 🧪', async () => {
+    await runCommandInProject(`npx cypress-expect --min-passing 5 --pending 0`, projectPath)
+  })
+
+  it('Run e2e skip tests without grep 🧪', async () => {
```

Comment:
I would expect these system tests to be modeled like the other system-tests in Cypress for maintainability purposes. If we need to make enhancements to support checking the report output, I think that'd be better than adding dozens of json files with the test results don't live with the test. At a min, this need to be running against the binary and not using `npx` in these tests.

Reply:
The issue is that the plugin needs the `cypress` module, and it can't resolve without  it. It is the same issue we have with `@cypress/schematics`. This is why we have to move the system tests to a temp dir to run

Reply:
@jordanpowell88 What do you mean exactly? Because it's listed as a peerDependency?

Reply:
Yes it needs cypress as a peerDep but it also needs cypress as a devDep

Reply:
Looking how this is setup though, that's because of `cypress-expect` right? which I why I am wondering if that's the right tool if we are unable to setup the system-tests to use the just-built binary. It looks like this is just a wrapper around running cypress programmatically to return the test results. We can pull the cypress output from `systemTests.it` to determine the tests that ran and how many passed & failed.

Reply:
Correct, we can do this but this would be a bigger lift than we are wanting to do at this moment